codeforpdx/PASS
# Add some scripts and environment config to allow developers to easily use a local server.

### Purpose:
This PR adds `community-solid-server` as a developer dependency to the project, and integrates it with the PASS application. The intent is to allow developers to run a solid server locally that they can test the PASS project against. That way they don't have to access `opencommons.net` to run tests. Although we can still use `opencommons.net` as an integration testing area, and developers can still target that as their server if they choose.

### Changes

- PASS now checks the runtime environment, instead of a hardcoded string, to get the pod URL.
- PASS checks a different env variable depending on the application `mode`
- `community-solid-server` is added as a dev dependency
- The script `dev:pod` is added to `package.json` to make starting the solid server easier

### Testing

1. Pull Branch
2. Run `npm install`
3. Run `npm run dev:pod`
4. Run `npm run dev`
5. Test that PASS points to localhost:3000, not opencommons.net. The application should work otherwise normally
6. Take down PASS server
7. Run `npm run build`
8. Check that PASS points to opencommons.net
